client/header/activity-panel/panels/inbox/card.js

```diff
@@ -39,39 +48,210 @@ class InboxNoteCard extends Component {
 		}
 	}
 
+	openDismissModal( type, onToggle ) {
+		this.setState( {
+			isDismissModalOpen: true,
+			dismissType: type,
+		} );
+		onToggle();
+	}
+
+	closeDismissModal() {
+		this.setState( {
+			isDismissModalOpen: false,
+		} );
+	}
+
+	handleBlur( event, onToggle ) {
+		const targetIsRenderedContent = event.relatedTarget
+			? event.relatedTarget.className.includes(
+					'woocommerce-admin-dismiss-notification'
+			  )
+			: false;
+		if ( targetIsRenderedContent ) {
+			event.preventDefault();
+		} else {
+			onToggle();
+		}
+	}
+
+	renderDismissButton() {
+		return (
+			<Dropdown
+				position="bottom right"
+				renderToggle={ ( { onToggle } ) => (
+					<Button
+						isTertiary
+						onClick={ onToggle }
+						onBlur={ ( event ) =>
+							this.handleBlur( event, onToggle )
+						}
+					>
+						{ __( 'Dismiss', 'woocommerce-admin' ) }
+					</Button>
+				) }
+				focusOnMount={ false }
+				popoverProps={ { noArrow: true } }
+				renderContent={ ( { onToggle } ) => (
+					<ul>
+						<li>
+							<Button
+								className="woocommerce-admin-dismiss-notification"
+								onClick={ () =>
+									this.openDismissModal( 'this', onToggle )
+								}
+							>
+								{ __(
+									'Dismiss this message',
+									'woocommerce-admin'
+								) }
+							</Button>
+						</li>
+						<li>
+							<Button
+								className="woocommerce-admin-dismiss-notification"
+								onClick={ () =>
+									this.openDismissModal( 'all', onToggle )
+								}
+							>
+								{ __(
+									'Dismiss all messages',
+									'woocommerce-admin'
+								) }
+							</Button>
+						</li>
+					</ul>
+				) }
+			/>
+		);
+	}
+
+	getDismissButton () {
```

Comment:
`getDismissButton`'s name doesn't seem right here, it's a bit confusing. Does this need to be a seperate function?